src/bscPlugin/fileProviders/ComponentStatementProvider.ts

```diff
@@ -56,26 +56,76 @@ export class ComponentStatementProvider {
+                    onChange: this.getAnnotationValue(member.annotations?.filter(x => x.name.toLowerCase() === 'onchange')),
+                    alwaysNotify: this.getAnnotationValue(member.annotations?.filter(x => x.name.toLowerCase() === 'alwaysnotify')) === 'true' ? 'true' : ''
+                };
+
+                let field = '<field';
+                Object.keys(fieldAttributes).forEach(attribute => {
+                    let value = fieldAttributes[attribute];
+                    // Only add a attribute if the value is not empty.
+                    if (value !== '') {
+                        field += ` ${attribute}="${fieldAttributes[attribute]}"`;
+                    }
+                });
+                field += ' />';
+                return field;
             } else {
                 return '';
             }
         }).filter(x => !!x);
 
+        let componentChildren = '';
+        const template = statement.annotations?.find(x => x.name.toLowerCase() === 'template');
+        if (template) {
+            // TODO: Better strip of component and children elements.
+            componentChildren = `<children>${this.getAnnotationValue([template]).replaceAll('<component>', '').replaceAll('</component>', '').replaceAll('<children>', '').replaceAll('</children>', '')}</children>`;
+        }
+
         xmlFile.parse(undent`
             <component name="${name}" extends="${statement.getParentName(ParseMode.BrightScript) ?? 'Group'}">
                 <script uri="${util.sanitizePkgPath(file.destPath)}" />
                 <script uri="${util.sanitizePkgPath(codebehindFile.destPath)}" />
                 ${interfaceMembers.length > 0 ? '<interface>' : ''}
                     ${interfaceMembers.join('\n                    ')}
                 ${interfaceMembers.length > 0 ? '</interface>' : ''}
+                ${componentChildren}
             </component>
         `);
 
-
         this.event.files.push(xmlFile);
     }
 
+    private getAnnotationValue(annotations: any) {
+        let response = [];
+        if (annotations !== undefined) {
+            annotations.forEach(a => {
+                let args = a?.call?.args[0];
+                if (isVariableExpression(args) || isDottedGetExpression(args)) {
+                    response.push(args.name.text);
+                } else if (isLiteralExpression(args)) {
+                    let values = args?.token?.text.replaceAll('\"', '').replaceAll(' ', '').split(',');
+                    response = response.concat(values);
+                } else if (isTemplateStringExpression(args)) {
+                    let textOutput = '';
+                    args.quasis[0]?.expressions?.forEach((a: { token: { text: string } }) => {
+                        textOutput += a.token.text;
+                    });
+                    response.push(textOutput);
+                }
+            });
```

Comment:
Let's use a for loop here instead, since it's slightly faster at scale
```suggestion
           for (const annotation of annotations) {
                let args = annotation?.call?.args[0];
                if (isVariableExpression(args) || isDottedGetExpression(args)) {
                    response.push(args.name.text);
                } else if (isLiteralExpression(args)) {
                    let values = args?.token?.text.replaceAll('\"', '').replaceAll(' ', '').split(',');
                    response = response.concat(values);
                } else if (isTemplateStringExpression(args)) {
                    let textOutput = '';
                    args.quasis[0]?.expressions?.forEach((a: { token: { text: string } }) => {
                        textOutput += a.token.text;
                    });
                    response.push(textOutput);
                }
            });
```